tests/functional/glusterd/test_glusterd_statedump_with_server_and_client_quorum_set.py

```diff
@@ -0,0 +1,332 @@
+                            " statedump")
+        g.log.info("Client's desciption is present as expected, in statedump")
+
+    def test_glusterd_statedump_when_quorum_set_on_volumes(self):
+        """
+        Test Case:
+        1. Create and start a volume
+        2. Enable quota on the volume
+        2. Fuse mount the volume
+        3. Get the glusterd statedump and analyze the statedump
+        4. Enable client-side quorum on the volume
+        5. Get the glusterd statedump
+        6. Delete the volume and peer detach 2 nodes
+        7. Create a replica 2 volume and start it
+        8. Kill the first brick of the volume
+        9. Get the glusterd statedump
+        10. Start the volume with force
+        11. Enable server-side quorum on the volume
+        12. Get the glusterd statedump
+        13. Stop glusterd on one of the node
+        14. Get the glusterd statedump
+        15. Stop glusterd on another node
+        16. Get the glusterd statedump
+        """
+        # pylint: disable=too-many-statements
+        # Enable Quota
+        ret, _, _ = quota_enable(self.mnode, self.volname)
+        self.assertEqual(ret, 0, ("Failed to enable quota on the volume %s",
+                                  self.volname))
+        g.log.info("Successfully enabled quota on the volume %s", self.volname)
+
+        # Mount the volume
+        ret, _, _ = mount_volume(self.volname, mtype=self.mount_type,
+                                 mpoint=self.mounts[0].mountpoint,
+                                 mserver=self.mnode,
+                                 mclient=self.mounts[0].client_system)
+        self.assertEqual(ret, 0, "Failed to mount the volume: %s"
+                         % self.volname)
+        g.log.info("Successfully mounted the volume: %s", self.volname)
+
+        # Get the statedump of glusterd process
+        self.dump_count = 1
+        self._get_statedump_of_glusterd(self.dump_count)
+
+        # Analyze the statedump created
+        self._analyze_statedump()
+
+        # Enable client-side quorum on volume
+        option = {"cluster.quorum-type": "auto"}
+        self._set_option_for_volume(self.volname, option)
+
+        # Get the statedump of glusterd process
+        self.dump_count += 1
+        self._get_statedump_of_glusterd(self.dump_count)
+
+        # Delete the volume
+        ret = cleanup_volume(self.mnode, self.volname)
+        self.assertTrue(ret, "Failed to delete the volume: %s" % self.volname)
+        g.log.info("Successfully deleted the volume: %s", self.volname)
+
+        # Peer detach two nodes
+        ret = peer_detach_servers(self.mnode, self.servers[1:3])
+        self.assertTrue(ret, "Failed to detach the servers %s"
+                        % self.servers[1:3])
+        g.log.info("Successfully detached peers %s", self.servers[1:3])
+
+        self.volume_config = {
+            'name': 'replica-volume',
```

Comment:
Add the testcase at the start of the volume name so that it doesn't cause any failures.

Reply:
Do you mean a comment before the volume declaration?

Reply:
@kshithijiyer in case you missed the above comment. Can you please confirm the above requirement?

Reply:
More like this
```
        self.volume_config = {
            'name': ('test_glusterd_statedump_when_quorum_'
                         set_on_volumes_replica-volume'),
```